src/core/agent/services/credentialService.ts

```diff
@@ -127,16 +135,50 @@ class CredentialService extends AgentService {
   async deleteCredential(id: string): Promise<void> {
     const metadata = await this.getMetadataById(id);
     this.validArchivedCredential(metadata);
-    // We only soft delete because we need to sync with KERIA. This will prevent re-sync deleted records.
+
     await this.credentialStorage.updateCredentialMetadata(id, {
-      isDeleted: true,
+      pendingDeletion: true,
     });
   }
 
   async deleteStaleLocalCredential(id: string): Promise<void> {
     await this.credentialStorage.deleteCredentialMetadata(id);
   }
 
+  async deleteCredentialOnKeria(id: string): Promise<void> {
+    const metadata = await this.getMetadataById(id);
+    const identifier = await this.props.signifyClient
+      .identifiers()
+      .get(metadata.identifierId);
+
+    if(metadata.pendingDeletion == true){
+      await this.props.signifyClient
+        .credentials()
+        .revoke(identifier.name, metadata.id)
+        .catch(async (error) => {
+          const status = error.message.split(" - ")[1];
+          if (/404/gi.test(status)) {
+            return  await this.deleteStaleLocalCredential(id)
+          } else {
+            throw error;
+          }
+        });
+
+      await this.deleteStaleLocalCredential(id)
+    }
+  }
+
+  async markCredentialPendingDelete(id: string) {
+    await this.deleteCredential(id);
```

Comment:
Can you move the contents of `deleteCredential` here? Otherwise the naming makes no sense, and there's no need for two separate functions.